### PR TITLE
Don't verify segment flatbuffers on startup

### PR DESCRIPTION
**Pull request by lava**
###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/vast/fbs/utils.hpp [hunk 1]
```diff
@@ -102,6 +102,12 @@ caf::error deserialize_bytes(const flatbuffers::Vector<Byte>* v, T& x) {
   return caf::none;
 }
 
+template <class Flatbuffer, size_t Extent = dynamic_extent>
```
Comment (mavam) [inline libvast/vast/fbs/utils.hpp:105]: Mind adding short documentation to this utility?

**Review by mavam — APPROVED** [review]
Looks good.

Please add some documentation to the utility and then feel free to merge.

**Comment by lava** [comment]
> EDIT: sorry, I overlooked that this is a draft PR. Ignore the merge part for now.

Maybe putting `wip: ` in the PR title has its uses after all ;p

**Review by mavam — APPROVED** [review]
LGTM. I didn't tested it again since you already reported the obvious performance numbers.

Optional: you could now call `as_unverified_flatbuffer` in the implementation of `as_flatbuffer`.

**Comment by dominiklohmann** [comment]
Not quite sure what value `as_unverified_flatbuffer<RootType>(as_bytes(chunk))` provides over `fbs::GetRootType(chunk.data())`.

File: libvast/vast/fbs/utils.hpp [hunk 2]
```diff
@@ -27,7 +30,10 @@
 
 #include <flatbuffers/flatbuffers.h>
 
+#include <algorithm>
 #include <cstdint>
+#include <ctime>
+#include <dirent.h>
```
Comment (dominiklohmann) [inline libvast/vast/fbs/utils.hpp:36]: What's up with the new includes? I don't think `fbs/utils.hpp` should include the accountant, or `type.hpp`, or `dirent.h`, whatever that is.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

**Comment by lava** [comment]
> Not quite sure what value as_unverified_flatbuffer<fbs::RootType>(as_bytes(chunk)) provides over fbs::GetRootType(chunk.data()).

Yeah, good point.

File: libvast/vast/fbs/utils.hpp [hunk 3]
```diff
@@ -109,7 +113,7 @@ caf::error deserialize_bytes(const flatbuffers::Vector<Byte>* v, T& x) {
 /// @param xs The buffer to unpack a flatbuffer from.
 /// @returns A pointer to the unpacked flatbuffer of type `Flatbuffer` or
 ///          `nullptr` if verification failed.
-/// @relates unverified_unpack
+/// @relates as_unverified_flatbuffer
```
Comment (dominiklohmann) [inline libvast/vast/fbs/utils.hpp:116]:
```suggestion
```

File: libvast/vast/fbs/utils.hpp [hunk 4]
```diff
@@ -27,6 +30,7 @@
 
 #include <flatbuffers/flatbuffers.h>
 
+#include <algorithm>
```
Comment (dominiklohmann) [inline libvast/vast/fbs/utils.hpp:33]:
```suggestion
```

File: libvast/vast/fbs/utils.hpp [hunk 5]
```diff
@@ -16,9 +16,12 @@
 #include "vast/byte.hpp"
 #include "vast/detail/type_traits.hpp"
 #include "vast/error.hpp"
+#include "vast/factory.hpp"
 #include "vast/fbs/version.hpp"
 #include "vast/fwd.hpp"
 #include "vast/span.hpp"
+#include "vast/system/accountant.hpp"
+#include "vast/type.hpp"
```
Comment (dominiklohmann) [inline libvast/vast/fbs/utils.hpp:24]:
```suggestion
```

File: libvast/vast/fbs/utils.hpp [hunk 6]
```diff
@@ -16,9 +16,12 @@
 #include "vast/byte.hpp"
 #include "vast/detail/type_traits.hpp"
 #include "vast/error.hpp"
+#include "vast/factory.hpp"
```
Comment (dominiklohmann) [inline libvast/vast/fbs/utils.hpp:19]:
```suggestion
```

File: libvast/src/segment_store.cpp [hunk 7]
```diff
@@ -351,10 +351,19 @@ caf::error segment_store::register_segment(const path& filename) {
   auto chk = chunk::mmap(filename);
   if (!chk)
     return make_error(ec::filesystem_error, "failed to mmap chunk", filename);
-  auto s = fbs::as_flatbuffer<fbs::Segment>(as_bytes(chk));
+  // We don't verify the segment here, since doing that would access
+  // most of the pages of the mapping and effectively cause us to
+  // read of the whole archive contents from disk. When the database
+  // approaches the terabyte range, this becomes prohibitively expensive.
+  // (see also dcso/ch1935)
```
Comment (dominiklohmann) [inline libvast/src/segment_store.cpp:358]:
```suggestion
  // (see also tdhtf/ch1935)
```

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

**Review by dominiklohmann — APPROVED** [review]
LGTM!

**Pull request merged**